schools_change_detection/schools_change_detection.py

```diff
@@ -0,0 +1,862 @@
+    facilities_name: Optional[str] = None
+    facilities_use: Optional[str] = None
+    facilities_subtype: Optional[str] = None
+    last_modified: Optional[date] = None
+    sql: Optional[str] = None
+
+    def get_geo_record(self) -> GeoRecord:
+        """
+        return the attributes of the school as a geo_record appropriate to be written to geopackage.
+        """
+        geo_record = {
+            "geometry": self.geom,
+            "properties": {
+                "facility_id": self.facilities_id,
+                "source_facility_id": self.source_id,
+                "name": self.facilities_name,
+                "source_name": self.source_name,
+                "use": self.facilities_use,
+                "use_type": self.source_type,
+                "use_subtype": self.facilities_subtype,
+                "estimated_occupancy": self.occupancy,
+                "last_modified": self.last_modified,
+                "change_action": self.change_action,
+                "change_description": self.change_description,
+                "comments": "",
+                "sql": self.sql,
+            },
+        }
+        return geo_record
+
+    def parse_check_results(self, check_results: Dict, moe_match: MOESchool) -> None:
+        """
+        Parse the analysis results
+        """
+        # PARSE REMOVE
+        ## change status to remove
+        if check_results["remove"]:
+            self.change_action = "remove"
+        else:
+            # PARSE UPDATE
+            ## change status to update where if any of the update related checks are false
+            if not all(
+                result for check, result in check_results.items() if check != "remove"
+            ):
+                self.change_action = "update"
+                if not check_results["attributes"]:
+                    self.sql = self.generate_update_sql(moe_match)
+
+                ## append description with each update that is needed
+                self.change_description = ", ".join(
+                    [
+                        check
+                        for check, result in check_results.items()
+                        if not result and check != "remove"
+                    ]
+                )
+
+    def generate_update_sql(self, moe_match: MOESchool) -> str:
+        """
+        Function called when there is a need for an attribution update.
+        The function creates an update sql statement containing the updated
+        attributes from the matched MOE school.
+        Intended to be run against the NZ Facilities DB.
+        """
+        sql = (
+            "UPDATE facilities_lds.nz_facilities "
+            f"SET name='{self.source_name}', source_name='{moe_match.source_name}', "
+            f"use_type='{moe_match.source_type}', estimated_occupancy={moe_match.occupancy}, "
+            f"last_modified='{date.today()}' WHERE facility_id={self.facilities_id} "
+            f"AND source_facility_id={self.source_id};"
+        )
```

Comment:
The preceding SQL should only update a column value where a difference was detected for the corresponding value in the new source data, rather than updating all columns.  The exception is for "name" and "source_name" to be updated at the same time.